src/callbacks.cpp

```diff
@@ -0,0 +1,201 @@
+#include "main.hpp"
+
+/**
+ * @brief Callback function for the control message.
+ *
+ * This function is called when a control message is received. It processes the
+ * message, activates the node based on the message, parses the payload, and
+ * saves the data in a Command struct.
+ *
+ * @param msg The control message received.
+ */
+void WaypointNode::callback_control(const interfaces::msg::Control &msg) {
+    // Ignore messages that don't target this node
+    if (msg.target_id != this->get_name()) {
+        RCLCPP_DEBUG(
+            this->get_logger(),
+            "WaypointNode::callback_control: Ignored control message not "
+            "targeted for this node");
+        return;
+    }
+
+    // Init node state
+    reset_node();
+
+    // Activate or deactivate node based on the message
+    if (msg.active != this->get_active()) {
+        if (msg.active) {
+            this->activate();
+        } else {
+            this->deactivate();
+        }
+    }
+
+    // If the node should be deactivated, don't parse the payload as it is only
+    // relevant in active mode
+    if (!this->get_active()) {
+        return;
+    }
+
+    nlohmann::json cmd_json;
```

Comment:
Add comment before this paragraph that this paragraph parses the json

Reply:
Fixed in https://github.com/THI-Drone/waypoint_package/pull/3/commits/5a49068ac86cdab12bbed671d77177ca7fee6d9a